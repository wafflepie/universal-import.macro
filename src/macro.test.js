const { default: pluginTester } = require("babel-plugin-tester");
const plugin = require("babel-plugin-macros");

pluginTester({
  plugin,
  snapshot: true,
  babelOptions: { filename: __filename },
  tests: [
    `
      import universalImport from "./macro"

      universalImport("./foo", "false")
    `,
    `
      import universalImport from "./macro"

      universalImport("./foo", "true")
    `,
    `
      import universalImport from "./macro"

      universalImport("./foo", "process.env.NODE_ENV === 'test'")
    `,
  ],
});
