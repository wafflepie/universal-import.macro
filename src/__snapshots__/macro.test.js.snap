// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`unknown plugin 1. unknown plugin: 1. unknown plugin 1`] = `

import universalImport from "./macro"

universalImport("./foo", "false")

      ↓ ↓ ↓ ↓ ↓ ↓

import("./foo");


`;

exports[`unknown plugin 2. unknown plugin: 2. unknown plugin 1`] = `

import universalImport from "./macro"

universalImport("./foo", "true")

      ↓ ↓ ↓ ↓ ↓ ↓

require("./foo");


`;

exports[`unknown plugin 3. unknown plugin: 3. unknown plugin 1`] = `

import universalImport from "./macro"

universalImport("./foo", "process.env.NODE_ENV === 'test'")

      ↓ ↓ ↓ ↓ ↓ ↓

require("./foo");


`;
